### PR TITLE
[RECONSTRUCTION] [Clang]Cleanup clang-analyzer warnings

### PR DESCRIPTION
**Pull request by smuzaffar**
This fixes clang static analyzer warnings https://cmssdt.cern.ch/SDT/jenkins-artifacts/ib-static-analysis/CMSSW_14_2_X_2024-10-02-1100/el8_amd64_gcc12/build-logs/

- remove dead code
- suppress warnings as
  - code is conditionally used by LogDebug/LogTrace

### CODE REVIEW

File: TrackPropagation/SteppingHelixPropagator/src/SteppingHelixPropagator.cc [hunk 1]
```diff
@@ -562,7 +563,6 @@ SteppingHelixPropagator::Result SteppingHelixPropagator::propagate(StateArray& s
         }
       } else {
         //keep this trial point and continue
-        dStep = defaultStep_;
```
Comment (slava77) [inline TrackPropagation/SteppingHelixPropagator/src/SteppingHelixPropagator.cc:565]: why is this removed?
this part is still inside `  while (makeNextStep) {`

Comment (slava77) [reply TrackPropagation/SteppingHelixPropagator/src/SteppingHelixPropagator.cc:565]: ah, nm, this value gets reset in the beginning of the while loop anyways

**Comment by smuzaffar** [comment]
please test

**Comment by jfernan2** [comment]
+1

**Comment by mandrenguyen** [comment]
+1
